ratelimits/limit.go

```diff
@@ -358,13 +386,65 @@ func (l *limitRegistry) getLimit(name Name, bucketKey string) (*Limit, error) {
 	return nil, errLimitDisabled
 }
 
+// loadOverrides replaces this registry's overrides with a new dataset. This is
+// separate from the goroutine in NewRefresher(), for ease of testing.
+func (l *limitRegistry) loadOverrides(ctx context.Context) error {
```

Comment:
The only caller has a `//nolint:errcheck`, which suggests that returning the error is pointless since nothing uses it.

However, I think this actually suggests a nicer solution: instead of logging from within `.loadOverrides`, simply return an error (appropriately prefixed), and let the caller log any errors from this function. That means collapsing the distinction between `Errf` and `Warning` but I think that's okay. Probably `Errf` is appropriate for both.

Reply:
In my first take at addressing this, I'm returning the error (and handling it in the caller) if `refreshOverrides` fails - I agree that's a lot nicer. But I'm still directly logging a warning if the set of overrides is empty, because that should be a non-fatal error, i.e. `NewTransactionBuilder` should still immediately succeed in that condition. Does that sound right?